app/controllers/api/v1/authentication_controller.rb

```diff
@@ -70,14 +70,12 @@ def send_pin
           return render_error(@mobile.errors.full_messages.join('. '))
         end
 
-        @user = User.find_by_mobile(@mobile.mobile)
-
-        if @user && @user.allowed_login?(app_name)
-          @user.send_verification_pin
-        elsif @user
+        @user = User.find_or_create_for_browse(is_browse_app?, @mobile.mobile)
```

Comment:
I think we should try to avoid having endpoints with different behaviours based on the app context (is_browse_app). It creates inconsistencies and makes it harder to test. Was talking about that recently with @steveyken

The name of the method should be explicit on what it does, "send_pin" should not secretly create a user.

I would be in favor of having a new method to the controller which does this "login or create user", without checking if it's the browse app. That way we can one day make use of it from another app without adding "or is_x_app"

Reply:
I agree 👍

Reply:
Yes, if we can start making the api less dependent on the calling app that will help us a lot in the future. (Good design principles). Agree re making a new endpoint. Something like `sign_up` would be a more suitable endpoint name.

Reply:
Yeah.. Good to add separate endpoint. Agree with it. 👍